engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -3356,23 +3362,30 @@ namespace dmGraphics
     static void VulkanSetTextureAsync(HTexture texture, const TextureParams& params)
     {
         // Async texture loading is not supported in Vulkan, defaulting to syncronous loading until then
-        SetTexture(texture, params);
+        VulkanSetTexture(texture, params);
     }
 
-    static void VulkanSetTextureParams(HTexture texture, TextureFilter minfilter, TextureFilter magfilter, TextureWrap uwrap, TextureWrap vwrap)
+    static float GetMaxAnisotrophyClamped(float max_anisotropy_requested)
     {
-        TextureSampler sampler = g_VulkanContext->m_TextureSamplers[texture->m_TextureSamplerIndex];
+        return dmMath::Min(max_anisotropy_requested, g_VulkanContext->m_PhysicalDevice.m_Properties.limits.maxSamplerAnisotropy);
+    }
+
+    static void VulkanSetTextureParams(HTexture texture, TextureFilter minfilter, TextureFilter magfilter, TextureWrap uwrap, TextureWrap vwrap, float max_anisotropy)
+    {
+        TextureSampler sampler   = g_VulkanContext->m_TextureSamplers[texture->m_TextureSamplerIndex];
+        float anisotropy_clamped = GetMaxAnisotrophyClamped(max_anisotropy);
 
-        if (sampler.m_MinFilter    != minfilter ||
-            sampler.m_MagFilter    != magfilter ||
-            sampler.m_AddressModeU != uwrap     ||
-            sampler.m_AddressModeV != vwrap     ||
-            sampler.m_MaxLod       != texture->m_MipMapCount)
+        if (sampler.m_MinFilter     != minfilter              ||
+            sampler.m_MagFilter     != magfilter              ||
+            sampler.m_AddressModeU  != uwrap                  ||
+            sampler.m_AddressModeV  != vwrap                  ||
+            sampler.m_MaxLod        != texture->m_MipMapCount ||
+            sampler.m_MaxAnisotropy != anisotropy_clamped)
```

Comment:
Maybe we should use ints instead of floats for the anisotrophy value everywhere. Technically the APIs support float values but I don't think non-integer values will do anything?

Reply:
Actually float could make sense it seems, but it's up to us anyway to dictate float/ints anyway:
https://stackoverflow.com/questions/57465803/why-opengl-use-float-to-define-the-level-of-anisotropic-filtering